src/java.base/share/classes/java/util/Currency.java

```diff
@@ -342,7 +343,8 @@ private static Currency getInstance(String currencyCode, int defaultFractionDigi
             if (!found) {
                 OtherCurrencyEntry ocEntry = OtherCurrencyEntry.findEntry(currencyCode);
                 if (ocEntry == null) {
-                    throw new IllegalArgumentException();
+                    throw new IllegalArgumentException("The input currency code" +
+                            " is not a supported ISO 4217 code");
```

Comment:
I think "valid" is more appropriate than "supported." Applies to other locations as well.